lib/iris/_data_manager.py

```diff
@@ -185,7 +199,15 @@ def _assert_axioms(self):
                 'real data and realised {!r}.')
         assert state, emsg.format(self._realised_dtype)
 
-    def _deepcopy(self, memo, data=None, realised_dtype='none'):
+        state = not (self.has_lazy_data() and
+                     self._lazy_array.dtype.kind != 'f' and
+                     self._realised_dtype is not None)
```

Comment:
I think this is wrong, but it is very hard to read !!!
I suspect it should say : `self._realised_dtype is None` ?

If I got it right, I think we want to check that ..
  * when we have lazy, non-floating data ..
  * .. it ***must*** have a realisation dtype.

In that case, the expression in brackets is "the thing we don't want" :
  * when.. we have lazy data
  * .. and the type is non-floating
  * .. and there is *no* realisation dtype (not as written)